yt/yt/server/controller_agent/controllers/task.cpp

```diff
@@ -1830,6 +1867,15 @@ void TTask::AddOutputTableSpecs(
     const auto& outputStreamDescriptors = joblet->OutputStreamDescriptors;
     YT_VERIFY(joblet->ChunkListIds.size() == outputStreamDescriptors.size());
     auto* jobSpecExt = jobSpec->MutableExtension(TJobSpecExt::job_spec_ext);
+    if (joblet->OutputCookieGroupIndex) {
+        SetProtoExtension<NChunkClient::NProto::TDataSourceDirectoryExt>(
+            jobSpecExt->mutable_extensions(),
+            New<TDataSourceDirectory>());
+        SetProtoExtension<NChunkClient::NProto::TDataSinkDirectoryExt>(
+            jobSpecExt->mutable_extensions(),
+            New<TDataSinkDirectory>());
+        return;
```

Comment:
Why `return` is written here?

Reply:
So there would be no `output_table_specs`

Reply:
I am worried about the fact that we add input stream descriptors below (but it may not an issue for this particalar PR)